adage/analyze/views.py

```diff
@@ -1,22 +1,42 @@
-from rest_framework.decorators import api_view
-from rest_framework.response import Response
-from rest_framework.reverse import reverse
 from rest_framework.viewsets import ReadOnlyModelViewSet
+from .models import Experiment, MLModel
+from .serializers import ExperimentSerializer, MLModelSerializer
 
-from .models import MLModel
-from .serializers import MLModelSerializer
 
+class ExperimentViewSet(ReadOnlyModelViewSet):
+    """Experiments viewset"""
+    http_method_names = ['get']
+    serializer_class = ExperimentSerializer
+
+    def get_queryset(self):
+        queryset = Experiment.objects.all()
+        # 'search' parameter, which does full text search based on the
+        # following 3 fields in Experiment model:
+        # - "accession"
+        # - "name"
+        # - "description"
+        search_str = self.request.query_params.get('search', None)
+        if search_str is not None:
+            from django.contrib.postgres.search import (
+                SearchQuery, SearchRank, SearchVector
+            )
```

Comment:
Would it be a problem to put these import statements at the top of `analyze/views.py`? I am not sure what we gain by doing the import conditionally down here. I generally like having all of the imports together because it's clear what the module's dependencies are.

Reply:
The main reason why I put these imports here is because they are Postgres-specific. When we need to support another database engine (such as `SQLite`) in the future, we will have to do these imports conditionally.

Reply:
I can see that the imports would need to be done conditionally depending upon which database engine is in use, but that's not what's happening here. These are being imported `if search_str is not None`, so the modules will be imported each time a search parameter is passed in.

Ultimately, yes, we will want to do a conditional import based upon which database engine is in use, but that should still happen at the top of this `analyze/views.py` module. (It makes sense to defer the conditional import logic until the `SQLite` implementation, though.)